jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/traveler/TravelerMessageController.java

```diff
@@ -1,56 +1,97 @@
-//        String timHex = "3081C68001108109000000000000003714830101A481AE3081AB800102A11BA119A0108004194FBA1F8104CE45CE2382020A0681020006820102820207DE830301C17084027D00850102A6108004194FC1988104CE45DA4082020A008702016E880100A92430228002000EA21CA01AA31804040CE205A104040ADA04F70404068004D60404034D0704AA3AA0383006A004800235293006A0048002010C3006A004800231283006A004800222113006A0048002010C3006A004800231203006A0048002221185021001";
-//        byte [] tim_ba = HexTool.parseHex(timHex, false);
-//        InputStream ins = new ByteArrayInputStream(tim_ba);
-//
-//        TravelerInformation tim = new TravelerInformation();
-//
-//        System.out.print(tim);
-//        try {
-//            coder.decode(ins, tim);
-//
-//        } catch (Exception e) {
-////            System.out.print( e);
-//        } finally {
-//            try {
-//                ins.close();
-//            } catch (IOException e) {
-//            }
-//        }
-//
-//        MsgCount cnt = new MsgCount(msgcnt);
-//        tim.setMsgCnt(cnt);
+        
+        
+        // Step 1 - Serialize the JSON into a TIM object
         TravelerSerializer timObject = new TravelerSerializer(jsonString);
-        System.out.print(timObject.getTravelerInformationObject());
+        
+        
+        // Step 2 - Populate the SnmpProperties object with SNMP preferences
+        JSONObject obj = new JSONObject(jsonString);
+        JSONArray rsuList = obj.getJSONArray("RSUs");
+        String ip = rsuList.getJSONObject(0).getString("target");
+        String user = rsuList.getJSONObject(0).getString("username");
+        String pass = rsuList.getJSONObject(0).getString("password");
+        int retries = Integer.parseInt(rsuList.getJSONObject(0).getString("retries"));
+        int timeout = Integer.parseInt(rsuList.getJSONObject(0).getString("timeout"));
+        
+        Address addr = GenericAddress.parse(ip + "/161");
+
+        SnmpProperties testProps = new SnmpProperties(addr, user, pass, retries, timeout);
+        logger.debug("TIM CONTROLLER - Serialized TIM: {}", timObject.getTravelerInformationObject().toString());
+        
+        // Step 2 - Encode the TIM object to a hex string
+        String rsuSRMPayload = null;
+        try {
+            rsuSRMPayload = timObject.getHexTravelerInformation();
+            if (rsuSRMPayload == null) {
+                throw new TimMessageException("Returned null string"); 
+            }
+        } catch (Exception e) {
+            logger.error("TIM CONTROLLER - Failed to encode TIM: {}", e);
+            return "{\"success\": false}";
+        }
+        logger.debug("TIM CONTROLLER - Encoded Hex TIM: {}", rsuSRMPayload);
+        
+        
+        // Step 3 - Populate the TimParameters object with OID values
+        JSONObject snmpParams= obj.getJSONObject("snmp");
 
+        String rsuSRMPsid = snmpParams.getString("rsuid");
+        int rsuSRMDsrcMsgId = Integer.parseInt(snmpParams.getString("msgid"));
```

Comment:
Please use snmpParam's getInt method instead of using getString and parsing it to integer.